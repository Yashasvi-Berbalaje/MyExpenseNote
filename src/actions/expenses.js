import {v4 as uuidv4} from 'uuid';

const AddExpense = ({
    description='',
    note='',
    amount=0,
    createdAt=0
}={}) =>({
    type:'ADD',
    expenses:{
       id:uuidv4(),
       description,
       note,
       amount,
       createdAt
    }
})

const removeExpense = ({id}={}) =>({
    type:'REMOVE',
    id
})

const EditExpense = (id,update) =>({
    type:'EDIT_EXPENSE',
    id,
    update
})

export {AddExpense,removeExpense,EditExpense};