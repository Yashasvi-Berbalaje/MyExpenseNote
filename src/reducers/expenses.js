
const expenseReducerDefaultState=[]
const expenseReducer = (state=expenseReducerDefaultState,action) => {
     switch(action.type){
        case 'ADD':
             return[...state,action.expenses];
        case 'REMOVE':
        return (state.filter(({id}) => id!==action.id));
        case 'EDIT_EXPENSE':
        return(
         state.map((expense)=>{
            if(expense.id===action.id)
            {
                
               return{...expense,...action.update} 
            }
            else{
                return expense;
            }
         })   
        )
         default:
             return state;
     }
}

export default expenseReducer;