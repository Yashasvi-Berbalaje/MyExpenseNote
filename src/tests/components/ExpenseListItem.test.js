import React from 'react';
import {shallow} from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import toJSON from 'enzyme-to-json';
import expenses from '../fixtures/expenses';

test('Test case for Expense List Item',()=>{
    const action = shallow(<ExpenseListItem {...expenses[1]}/>)
    expect(toJSON(action)).toMatchSnapshot();
})
