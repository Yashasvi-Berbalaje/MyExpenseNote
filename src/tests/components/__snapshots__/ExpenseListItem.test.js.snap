// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test case for Expense List Item 1`] = `
<div>
  <Link
    to="/edit/2"
  >
    <p>
      Rent
    </p>
  </Link>
  <p>
    10000
  </p>
  <p>
    -345600000
  </p>
</div>
`;
